src/Application/Console/Formatters/Json.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Application\Console\Formatters;
+
+use Exception;
+use InvalidArgumentException;
+use NunoMaduro\PhpInsights\Application\Console\Contracts\Formatter;
+use NunoMaduro\PhpInsights\Domain\Contracts\HasDetails;
+use NunoMaduro\PhpInsights\Domain\Insights\Insight;
+use NunoMaduro\PhpInsights\Domain\Insights\InsightCollection;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @internal
+ */
+final class Json implements Formatter
+{
+    /** @var OutputInterface */
+    private $output;
+
+    public function __construct(InputInterface $input, OutputInterface $output)
```

Comment:
Need to double check this, but maybe we can improve this so our formatters are not depending of the `Symfony\Component\Console` layer.

Reply:
Right now they are depending on it beca the console layer has the `press to continue` logic.
We could get around it by creating a class which gives this possibility. However not really sure if I think it would be an advantage.